kafka/pykafka/consumer.py

```diff
@@ -47,9 +199,63 @@ class OwnedPartition(object):
     Used to keep track of offsets and the internal message queue.
     """
 
-    def __init__(self, partition):
+    def __init__(self, partition, consumer):
         self.partition = partition
+        self.consumer = consumer
         self._messages = Queue()
+        self.last_offset_consumed = 0
+        self.next_offset = 0
+
+        if self.consumer._auto_commit_enable and self.consumer.consumer_group is not None:
+            self.last_offset_consumed = self._fetch_last_known_offset()
 
     def consume(self, timeout=None):
+        """Get a single message from this partition
+        """
+        if self._messages.empty():
+            self._fetch(timeout=timeout)
+
+        try:
+            message = self._messages.get_nowait()
+            self._queued_chunks -= 1
```

Comment:
Definitely check with @yungchin because i think it also has min_messages_queued.  That way, there's a window in there where the queue is mostly full, but not entirely full, and you don't ask for more messages.  This avoids the case where every single read of the queue causes a request to the broker.  Pretty sure librdkafka does that. If not, we won't do it either because it'd be difficult to do on the librdkafka side.